### PR TITLE
chore: Quality pass on testing chapter

### PR DESCRIPTION
**Pull request by bebraw**

### CODE REVIEW

File: manuscript/code-quality/04-testing.md [hunk 1]
```diff
@@ -113,22 +113,21 @@ Good property testing tools like [JSVerify](https://jsverify.github.io/) or [tes
...
 
 ### Regression Testing
 
 **Regression testing** makes sure the software works still the same way as before without breaking anything.
- Regressions can exist on multiple levels, and regression testing complements the techniques discussed earlier.
 
- **Approval testing** is a specific technique that allows you to develop regression tests against an existing system. The idea is to store existing system state first as **snapshots** and then use this data for testing that behavior doesn't change while the implementation is being refactored.
+Regressions can exist on multiple levels, and regression testing complements the techniques discussed earlier.
+
+**Approval testing** is a specific technique that allows you to develop regression tests against an existing system. The idea is to store existing system state first as **snapshots** and then use this data for testing that behavior doesn't change while the implementation is being refactored.
```
Comment (sapegin) [inline manuscript/code-quality/04-testing.md]: ”is a specific technique that” → ””.

File: manuscript/code-quality/04-testing.md [hunk 2]
```diff
@@ -169,11 +165,11 @@ Tools like [complexity-report](https://www.npmjs.com/package/complexity-report),
 
 ### Smoke Testing
 
-TODO: See how projects like Node do this
+The idea of *smoke testing* is to verify that vital functionality works in production. The purpose of these tests is to give high level idea of whether the application might be working or not. They are light to write and can be a starting point if the codebase doesn't have any tests yet. You could for instance assert that the application runs correctly and then expand around that definition as you identify functionality like this.
```
Comment (sapegin) [inline manuscript/code-quality/04-testing.md]: What ”runs correctly” means in this context? I’d give a concrete example, like *log in and log out after each production deploy* (that’s what we do as a smoke test). But it doesn’t test that the app runs correctly and I think the purpose of some tests is to test that the app runs at all.

“then expand around that definition as you identify functionality like this” → “” ;-)

File: manuscript/code-quality/04-testing.md [hunk 3]
```diff
@@ -169,11 +165,11 @@ Tools like [complexity-report](https://www.npmjs.com/package/complexity-report),
...
 
 ### Code Size
 
-TODO: https://www.npmjs.com/package/bundlesize, https://www.npmjs.com/package/size-limit
+The size of the code can be important especially for frontend related packages as you want to minimize the amount of code the client has to download and parse. Tools like [bundlesize](https://www.npmjs.com/package/bundlesize) and [size-limit](https://www.npmjs.com/package/size-limit) allow you to check against size through *package.json* configuration and a command line tool.
```
Comment (sapegin) [inline manuscript/code-quality/04-testing.md]: “against size through *package.json* configuration and a command line tool” → “to do that” or something, probably unnecessary details here.

File: manuscript/code-quality/04-testing.md (hunk 1 above)
Comment (rstegg) [reply manuscript/code-quality/04-testing.md]: "The idea is to store existing system state first as **snapshots** and then use this data for testing that behavior doesn't change while the implementation is being refactored" → 

"**Approval testing** first stores existing system state as **snapshots** and uses this data for testing a behavior for changes while the implementation is being refactored"

Discuss this change?

**Comment by bebraw** [comment]
Let's go with this and do another pass later. I guess testing pyramids (at least two) should go here and we might want to split up the chapter somehow. Maybe based on theme.

**Pull request merged**